### ISSUE TITLE
HIGH: [Public] [$500] Members disappear after invitation in public rooms

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#36378]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.40.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Create a workspace
2. Create a public room
3. Invite a new member (without an Expensify account)
4. Observe the member appearing momentarily and then disappearing
5. Refresh the page and observe the newly invited member on the member list

## Expected Result:
The invited member shouldn't flash and then disappear

## Actual Result:
The members disappear 

## Workaround:
Unknowns

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38236554/0f83941a-6767-4a47-9702-f6535a0fa6b9

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016a8e93511c34c648</li>
        <li>Upwork Job ID: 1757156885762093056</li>
        <li>Last Price Increase: 2024-02-26</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related #wave8-collect-admins
CC @zanyrenney

**Comment by bfitzexpensify** [comment]
I think this likely fits better with #vip-vsb. I want to test this a little more tomorrow myself to determine priority.

**Comment by bfitzexpensify** [comment]
Raised in the #vip-vsb room [here](https://expensify.slack.com/archives/C066HJM2CAZ/p1708492074949839)

**Comment by bfitzexpensify** [comment]
Added this to the #vip-vsb project, and split out the two issues - keeping the members-disappearing sub-issue here, and taking the emails not being delivered to https://github.com/Expensify/App/issues/37067

**Comment by brunovjk** [comment]
I believe there is an irregularity with the api `InviteToRoom` response. When we invite to a `non-public room`, the response is complete:
<details>
<summary>'InviteToRoom' response to a non-public room</summary>


![image](https://github.com/Expensify/App/assets/95647348/b1a60674-ec14-4a84-923c-59f7f7c4c91c)

</details>

However, when we invite to a `public room`, the backend response seems incomplete to me:
<details>
<summary>'InviteToRoom' response to a public room</summary>


![image](https://github.com/Expensify/App/assets/95647348/3c66e5a1-6773-4d0a-a84f-886720bedb66)

</details>

Offline works normally.

**Comment by bfitzexpensify** [comment]
@marcochavezf do you have some context on this? Looks like you were working on some `InviteToRoom` items in https://github.com/Expensify/App/issues/30919

**Comment by wildan-m** [comment]
Hi, I have a proposal posted [here](https://github.com/Expensify/App/issues/37192#issuecomment-1963288279). It's about personalDetails filtering. Currently, we do not display members with no personal details. If we want to show them, we should consider my alternative proposal to remove the personalDetails null check in `RoomMembersPage`.

**Comment by bfitzexpensify** [comment]
Thanks for checking @fedirjh - added the `internal` label

**Comment by bfitzexpensify** [comment]
Waiting for a volunteer on this one

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by bfitzexpensify** [comment]
I'm still able to reproduce so I think this is still a valid issue:

https://github.com/Expensify/App/assets/38236554/05939612-1ef6-4fb4-a387-c994a198ad4b

**Comment by bfitzexpensify** [comment]
Still waiting for a volunteer

**Comment by bfitzexpensify** [comment]
Same status - assigned to a project and with the appropriate status, we're just waiting for a volunteer.

**Comment by bfitzexpensify** [comment]
Same update

**Comment by quinthar** [comment]
Am I correct in understanding this is a real reproducible issue, and it's front-end so should be doable with contributors, but there are no proposals?  What are we doing to get proposals?

**Comment by wildan-m** [comment]
@quinthar ideally it should be fixed in backend, but if you're looking for a frontend fix, you might want to consider my proposal.

https://github.com/Expensify/App/issues/36378#issuecomment-1963408169

**Comment by bfitzexpensify** [comment]
Yes, @wildan-m is right - the ideal fix here is BE, which is why it's labelled as `Internal` and we're not actively looking for external proposals.

**Comment by bfitzexpensify** [comment]
Still awaiting an internal volunteer. Shared in the #vip-vsb update so it's visible for someone looking for something.

**Comment by bfitzexpensify** [comment]
Same update, still waiting on a volunteer.

**Comment by bfitzexpensify** [comment]
Same update

**Comment by bfitzexpensify** [comment]
Still awaiting an internal volunteer

**Comment by bfitzexpensify** [comment]
Same status - awaiting internal engineering. This issue is still being highlighted in the #vip-vsb Spring 24 items that need to be completed, so it's not going unnoticed.

**Comment by brunovjk** [comment]
> Still awaiting an internal volunteer

**Comment by bfitzexpensify** [comment]
Same status - awaiting internal engineering.

**Comment by bfitzexpensify** [comment]
No change here.

**Comment by bfitzexpensify** [comment]
Closing in favour of https://github.com/Expensify/App/issues/40459, which Youssef is assigned to. I added the details of reproduction and the likely cause from this issue into that issue.
